numpy/core/shape_base.py

```diff
@@ -187,22 +178,105 @@ def atleast_3d(*arys):
     [[[1 2]]] (1, 1, 2)
 
     """
-    res = []
-    for ary in arys:
-        ary = asanyarray(ary)
-        if ary.ndim == 0:
-            result = ary.reshape(1, 1, 1)
-        elif ary.ndim == 1:
-            result = ary[_nx.newaxis, :, _nx.newaxis]
-        elif ary.ndim == 2:
-            result = ary[:, :, _nx.newaxis]
-        else:
-            result = ary
-        res.append(result)
-    if len(res) == 1:
-        return res[0]
-    else:
-        return res
+    return _unpack([atleast_nd(atleast_nd(a, 2, 0), 3, -1) for a in arys])
+
+
+def _atleast_nd_dispatcher(ary, ndim, pos=None):
+    return (ary,)
+
+
+@array_function_dispatch(_atleast_nd_dispatcher)
+def atleast_nd(ary, ndim, pos=0):
```

Comment:
In numpy.array you can pre-pend ones to the array-shape by giving a 'ndmin' parameter.
Therefore, I think the variable name 'ndim' should be renamed to 'ndmin' to make it consistent with numpy.array.

Reply:
I think the atleast_nd should allow multiple inputs in the same way as atleast_1d, atleast_2d and atleast_3d do. This will it easier to use since then the call syntax and output will be the same. Any written code with multiple uses of the function will be shorter like this:

    x1, x2, x3 = numpy.atleast_nd(x1,x2,x3, ndmin=3)